pkg/vault/contracts/TimelockAuthorizer.sol

```diff
@@ -96,14 +102,39 @@ contract TimelockAuthorizer is IAuthorizer, IAuthentication {
      */
     event PermissionRevoked(bytes32 indexed actionId, address indexed account, address indexed where);
 
+    /**
+     * @dev Emitted when a new root is set
+     */
+    event RootSet(address indexed root);
+
     constructor(address admin, IAuthentication vault) {
+        root = admin;
         _vault = vault;
-        GRANT_ACTION_ID = getActionId(TimelockAuthorizer.grantPermissions.selector);
-        REVOKE_ACTION_ID = getActionId(TimelockAuthorizer.revokePermissions.selector);
+
+        bytes32 grantActionId = getActionId(TimelockAuthorizer.grantPermissions.selector);
+        _grantPermission(getActionId(grantActionId, WHATEVER), admin, EVERYWHERE);
+
+        bytes32 revokeActionId = getActionId(TimelockAuthorizer.revokePermissions.selector);
+        _grantPermission(getActionId(revokeActionId, WHATEVER), admin, EVERYWHERE);
```

Comment:
> The goal of this to save `root` from having to manage the grant permission in order to then grant permissions to others, correct?

> I'd consider removing this so that the entire new permissions structure is contained within the migration contract. We might otherwise have situations where e.g. the migration is made root and at the end transfers root to a multisig, but we forget to transfer these permissions the migration would end up with.

Not sure how you can do this. I mean, someone needs to have permissions to grant others, if we create the authorizer without setting the root in the constructor then who is able to grant permissions?